jvb/src/main/kotlin/org/jitsi/videobridge/cc/BandwidthProbing.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.cc
+
+import org.jitsi.nlj.rtp.bandwidthestimation.BandwidthEstimator
+import org.jitsi.nlj.util.Bandwidth
+import org.jitsi.utils.concurrent.PeriodicRunnable
+import org.jitsi.utils.logging.DiagnosticContext
+import org.jitsi.utils.logging.TimeSeriesLogger
+import org.jitsi.videobridge.cc.allocation.BitrateControllerStatusSnapshot
+import org.jitsi.videobridge.cc.config.BandwidthProbingConfig
+import org.json.simple.JSONObject
+import java.util.function.Supplier
+import kotlin.random.Random
+
+class BandwidthProbing(
+    private val probingDataSender: ProbingDataSender,
+    private val statusSnapshotSupplier: Supplier<BitrateControllerStatusSnapshot>
+) : PeriodicRunnable(config.paddingPeriodMs), BandwidthEstimator.Listener {
+
+    /**
+     * The sequence number to use if probing with the JVB's SSRC.
+     */
+    private var seqNum = Random.nextInt(0xFFFF)
+
+    /**
+     * The RTP timestamp to use if probing with the JVB's SSRC.
+     */
+    private var ts: Long = Random.nextLong().and(0xFFFFFFFFL)
```

Comment:
Java code did `nextInt()` not `nextLong()` - since this is only a 32-bit value, `nextInt()` should be sufficient, and the `and` will force it to be a positive `Long`.

Reply:
Yeah, but then I can't `and` that value with `0xFFFFFFFFL` :/.  But I just remembered we have an `Int.toPositiveLong()` helper, so I'll use `Random.nextInt().toPositiveLong()`